### DISCUSSION TITLE
关于MessageSegment.markdown如何使用

### DISCUSSION THREAD

**Discussion by STESmly** [nonebot/discussions#501] [问答]
我在QQ开放平台给机器人申请开通了markdown模版的权限，但我不知道如何使用，如果直接复制粘贴官方的文档的话会报错这个是

**Comment by STESmly** [comment]
这个是官方文档链接：https://bot.q.qq.com/wiki/develop/api/openapi/message/post_markdown_messages.html

**Reply by yanyongyu** [reply]
```python
from nonebot.adapters.qq.models import MessageMarkdown, MessageMarkdownParams

MessageSegment.markdown(
	MessageMarkdown(
		template_id=xxx,
		params=[MessageMarkdownParams(key=xxx, value=xxx)]
	)
)
```

**Reply by STESmly** [reply]
nonebot.adapters.qq.exception.ActionFailed: <ActionFailed: 400, code=40011000, message=请求数据异常, trace_id=051e1380e08663413d56767b3ff6ae10>      这个是只能把自定义模版审核通过了才能使用对吧，只是申请了这个权限是会报错的？

**Reply by yanyongyu** [reply]
当然要模板通过审核了才能使用，自定义模板要使用custom template id

**Reply by STESmly** [reply]
> 当然要模板通过审核了才能使用，自定义模板要使用custom template id

我这边审核通过了个按钮消息，但
MessageSegment.keyboard(
        MessageKeyboard(
        id="xxx"
    )
报错
nonebot.adapters.qq.exception.ActionFailed: <ActionFailed: 500, code=304061, message=invalid content, trace_id=575d73d5a91bd81d9a20a1eb86e309c4>

是我格式没写对吗

**Reply by yanyongyu** [reply]
keyboard 必须和 markdown 同时使用

**Reply by STESmly** [reply]
> keyboard 必须和 markdown 同时使用
但群里面的灿酱说可以发纯按钮啊，因为我dua没达标发不了md

**Reply by MRSourLemon** [reply]
markdown审核通过了，报错仍为<nonebot.adapters.qq.exception.ActionFailed: <ActionFailed: 400, code=40011000, message=请求数据异常, trace_id=74f9e6f2893304d21a213690007c271f>
from nonebot.adapters.qq.models.common import MessageMarkdown, MessageMarkdownParams,MessageKeyboard

markdown = on_command("md", rule=to_me(), priority=10, block=True)

@markdown.handle()
async def markdown_method(event: Event):
    # 创建 MessageMarkdown 对象
    markdown_data = MessageMarkdown(
        template_id="102801649_1757943376",  # 模板 ID
        params=[
            MessageMarkdownParams(key="title", values=["value1"]),
            MessageMarkdownParams(key="title2", values=["value2"]),
            MessageMarkdownParams(key="content1", values=["content1"]),
            MessageMarkdownParams(key="content2", values=["content2"]),
            MessageMarkdownParams(key="content3", values=["content3"]),
        ]
    )
    markdown_seg = MessageSegment.markdown(markdown_data)
    await markdown.finish(markdown_seg )
    消息模板md是这么发送吗

**Reply by yanyongyu** [reply]
被动 markdown 需要 DAU 才能开通使用。另外你的用法也不正确，应该是 `custom_template_id`。

**Reply by MRSourLemon** [reply]
> markdown审核通过了，报错仍为<nonebot.adapters.qq.exception.ActionFailed: <ActionFailed: 400, code=40011000, message=请求数据异常, trace_id=74f9e6f2893304d21a213690007c271f> from nonebot.adapters.qq.models.common import MessageMarkdown, MessageMarkdownParams,MessageKeyboard
> 
> markdown = on_command("md", rule=to_me(), priority=10, block=True)
> 
> @markdown.handle() async def markdown_method(event: Event): # 创建 MessageMarkdown 对象 markdown_data = MessageMarkdown( template_id="102801649_1757943376", # 模板 ID params=[ MessageMarkdownParams(key="title", values=["value1"]), MessageMarkdownParams(key="title2", values=["value2"]), MessageMarkdownParams(key="content1", values=["content1"]), MessageMarkdownParams(key="content2", values=["content2"]), MessageMarkdownParams(key="content3", values=["content3"]), ] ) markdown_seg = MessageSegment.markdown(markdown_data) await markdown.finish(markdown_seg ) 消息模板md是这么发送吗

看了一遍文档 md所有公域机器人都可以申请 但是发送需要DAU达到2000（近一个月每日平均上行消息）才能申请金牌机器人 金牌及更高级机器人才能发送md及消息按钮 所以DAU小于2000的机器人可以申请模板但是无法发送 发送时貌似就会触发这个异常

**Reply by MRSourLemon** [reply]
> 被动 markdown 需要 DAU 才能开通使用。另外你的用法也不正确，应该是 `custom_template_id`。

疏忽了 你说的是对的 改为custom_templete_id后报错就正常了（nonebot.adapters.qq.exception.ActionFailed: <ActionFailed: 500, code=304041, message=reply message not allow markdown, trace_id=c64274b706bb2250bfbec885a0f72fd1>）templete_id貌似是自定义模板的id

**Reply by xinvxueyuan** [reply]
tip:讨论有结果请标记为 `因已过时关闭` 或 `因已解决关闭` ，谢谢！
